src/test/scala/com/ubirch/filter/services/kafka/FilterServiceIntegrationTestWithoutCache.scala

```diff
@@ -117,62 +115,64 @@ class FilterServiceIntegrationTestWithoutCache extends TestBase with EmbeddedRed
 
       withRunningKafka {
 
-        logger.info("trying stop redis")
         try {
           redis.stop()
         } catch {
           case _: Throwable =>
         }
 
-        val msgEnvelope = generateMessageEnvelope()
+        val msgEnvelope = generateMsgEnvelope()
 
         val (_, conf) = startKafka(bootstrapServers)
 
         //publish message first time
         publishToKafka(readConsumerTopicHead(conf), msgEnvelope)
-
         Thread.sleep(3000)
 
         consumeFirstMessageFrom[MessageEnvelope](readProducerForwardTopic(conf)).ubirchPacket.getUUID mustBe
           msgEnvelope.ubirchPacket.getUUID
-
         val cacheError1 = consumeFirstMessageFrom[Error](readProducerErrorTopic(conf))
         cacheError1.error mustBe "RedisIOException"
         cacheError1.causes mustBe List("unable to make cache lookup 'null'.")
+        val cacheError2 = consumeFirstMessageFrom[Error](readProducerErrorTopic(conf))
+        cacheError2.error mustBe "RedisIOException"
+        cacheError2.causes mustBe List("unable to add value for hash ODkzMTk= to cache.")
 
       }
     }
 
-    "not change default behaviour" in {
+    "not change default behaviour when nothing is found in cassandra" in {
       implicit val kafkaConfig: EmbeddedKafkaConfig =
         EmbeddedKafkaConfig(kafkaPort = PortGiver.giveMeKafkaPort, zooKeeperPort = PortGiver.giveMeZookeeperPort)
       val bootstrapServers = "localhost:" + kafkaConfig.kafkaPort
 
       withRunningKafka {
 
         try {
-          "fuser -k 6379/tcp" !
+          redis.stop()
```

Comment:
Is there any reason that this `redis` object is defined globally?
I don't see the benefit. just make it difficult to see the current state of this object.
I think a RedisServer object should be defined in each test case.

Reply:
Ah, it may be because of `EmbeddedRedis` ?